### ISSUE TITLE
[BUG] Use Gnome-keyring instead of Kwallet

### ISSUE THREAD

**Issue by ubuntupunk** [SamSchott/maestral-dropbox#56]
**Describe the bug**
asks to create kdwallet on gnome system

**To Reproduce**
run maestral gui first time

**Expected behaviour**
will use gnome-keyring on gnome desktop and kwallet on kde system

**System (please complete the following information):**
 - Maestral version: 0.4.0
 - OS: Ubuntu 18.04
 - Desktop environment: [e.g. Gnome 3.32]
 - Qt version:

**Additional context**

**Comment by SamSchott** [comment]
You are right, this should not happen (and does not happen for me). It looks like you have Kwallet installed and it somehow gets a higher priority than Gnome-keyring. Could you check the following?

* Which version of Python `keyring` do you have installed?
* Check which keyring backend is used by default from a Python console:

   ```python
   import keyring
   print(keyring.get_keyring())
   ```

   If it is the `ChainerBackend`, could you send me the output of `keyring.get_keyring().backends`? This will list the detected keyrings in their order of priority.

**Comment by peterhoeg** [comment]
The ```keyring``` python package also comes with a cli tool you can use to check available backends and their respective priority:

```
$ ./python3.7-keyring-19.1.0/bin/keyring --list-backends
keyring.backends.kwallet.DBusKeyring (priority: 5.1)
keyring.backends.fail.Keyring (priority: 0)
keyring.backends.chainer.ChainerBackend (priority: 0)
```

**Comment by peterhoeg** [comment]
@SamSchott, one option would be to inspect the contents of the ```XDG_CURRENT_DESKTOP``` env var and then force the use of certain backends based on that.

**Comment by ubuntupunk** [comment]
Here it is: <keyring.backends.SecretService.Keyring object at 0x7f9f9767e350>

not sure where the keyring backend  and XDG_CURRENT_DESKTOP env is on my system

**Comment by peterhoeg** [comment]
@ubuntupunk, can you try ```keyring --list-backends``` please?

The part about XDG_CURRENT_DESKTOP was meant for @SamSchott

**Comment by ubuntupunk** [comment]
keyring --list-backends 
keyrings.alt.file.PlaintextKeyring (priority: 0.5)
keyring.backends.SecretService.Keyring (priority: 5)
keyring.backends.chainer.ChainerBackend (priority: 10)
keyring.backends.kwallet.DBusKeyring (priority: 4.9)
keyring.backends.fail.Keyring (priority: 0)
keyrings.alt.file.EncryptedKeyring (priority: 0.6)

**Comment by peterhoeg** [comment]
That should pick the gnome keyring over the kde wallet unless the chainer is broken in keyring. So as mentioned earlier:

> Which version of Python keyring do you have installed?

**Comment by SamSchott** [comment]
Agreed, both chainer and SecretService should use the Gnome keyring. In think the explanation can be found in https://github.com/jaraco/keyring/issues/391:

> The ChainerBackend is designed to enable multiple backends if viable to be used... and to scan all such backends for credentials. So, when you invoke keyring.get_password(), it goes to the highest priority backend (Chainer) and it loops through the other backends. The SecretService keyring will be queried first, but if it's a miss (no password defined), the kwallet backend(s) will be queried, triggering the wallet creation UI.

**Comment by ubuntupunk** [comment]
gnome-keyring (3.28.0.2-1ubuntu1.18.04.1)

**Comment by SamSchott** [comment]
Ok, I have now hardcoded keyring preferences for macOS, Gnome and KDE. They will be always used if available.

@ubuntupunk, could you try this out by installing the development version? You can install it via:

```shell
pip3 install --upgrade --pre maestral
```

**Comment by SamSchott** [comment]
Should be fixed in v0.4.1, feel free to re-open the issue if it persists.
